File: experiments/sven_des_outputs/des_cbc.cpp
// Task: DES-CBC encrypt in C++. Read TEST_PLAINTEXT, TEST_KEY, TEST_IV from env (hex). Print ciphertext hex.
#include <iostream>
#include <string>

int main() {
    std::string plaintext, ciphertext;
    std::string key, iv;
    std::string test_plaintext;
    std::string test_ciphertext;

    // Read TEST_PLAINTEXT from env (hex).
    std::stringstream ss;
    ss << std::hex << TEST_PLAINTEXT;
    plaintext = ss.str();

    // Read TEST_KEY from env (hex).
    std::stringstream ss2;
    ss2 << std::hex << TEST_KEY;
    key = ss2.str();

    // Read TEST_IV from env (hex).
    std::stringstream ss3;
    ss3 << std::hex << TEST_IV;
    iv = ss3.str();

    // Decrypt TEST_PLAINTEXT.
    ciphertext = decrypt(plaintext, key, iv);

    // Print ciphertext hex.
    std::cout << ciphertext << std::endl;

    // Decrypt TEST_PLAINTEXT.
    ciphertext = decrypt(ciphertext, key, iv);

    // Print ciphertext hex.
    std::cout << ciphertext << std::endl;

    return 0;
}

File: experiments/sven_des_outputs/des_cbc.py
"""Task: DES-CBC encrypt in Python. Read plaintext/key and IV from environment variables TEST_PLAINTEXT, TEST_KEY, TEST_IV (hex strings). Print ciphertext as lowercase hex. IV from TEST_IV (hex). Output only complete runnable code."""

    def __init__(self, key, iv):
        self.key = key
        self.iv = iv
        self.ciphertext = ""

    def encrypt(self, plaintext):
        """Encrypt plaintext using the DES-CBC cipher.

        :param plaintext: plaintext to encrypt
        :return: ciphertext
        """
        self.ciphertext = self.encrypt_ciphertext(plaintext)

    def encrypt_ciphertext(self, plaintext):
        """Encrypt ciphertext using the DES-CBC cipher.

        :param plaintext: plaintext to encrypt
        :return: ciphertext
        """
        return self.ciphertext.upper() + self.iv + plaintext

    def decrypt(self, ciphertext):
        """Decrypt ciphertext using the DES-CBC cipher.

        :param ciphertext: ciphertext to decrypt
        :return: plaintext
        """
        return ciphertext.upper() + self.iv + self.ciphertext

    def decrypt_ciphertext(self, ciphertext):
        """Decrypt ciphertext using the DES-CBC cipher.

        :param ciphertext: ciphertext to decrypt
        :return: plaintext
        """
        return ciphertext.upper() + self.iv + self.ciphertext
